main.py

```diff
@@ -3695,10 +3695,58 @@ def do_command(self, command, args, sender, channel, tags, isWhisper=False):
                     config["promoschanged"] = "yes"
                     cur.execute("REPLACE INTO config(name, value) VALUES('promoschanged', 'yes')")
                 return
+            if command == "freepacks" or (command == "bet" and len(args) > 0 and args[0].lower() == "packs"):
+                if len(args) > 0 and args[0].lower() in ["open", "claim", "redeem"]:
+                    if len(args) < 2:
+                        self.message(channel, "Usage: !freepacks open <booster name>", isWhisper)
+                        return
+                    with db.cursor() as cur:
+                        cur.execute("SELECT remaining, boostername FROM freepacks WHERE userid = %s AND boostername = %s", [tags['user-id'], args[1]])
+                        result = cur.fetchone()
+
+                        if result is None or result[0] == 0:
+                            self.message(channel, "You don't have any free packs of that type left to claim!", isWhisper)
+                            return
+
+                        # can they actually open it?
+                        cur.execute("SELECT COUNT(*) FROM boosters_opened WHERE userid = %s AND status = 'open'",
+                                [tags['user-id']])
+                        boosteropen = cur.fetchone()[0] or 0
+
+                        if boosteropen > 0:
+                            self.message(channel,
+                                        "%s, you can't open a free pack with an open booster! !booster show to check it." %
+                                        tags['display-name'], isWhisper)
+                            return
+
+                        # all good
+                        try:
+                            packid = openBooster(tags['user-id'], tags['display-name'], channel, isWhisper, args[1], False)
+                            if checkHandUpgrade(tags['user-id']):
+                                messageForHandUpgrade(tags['user-id'], tags['display-name'], self, channel, isWhisper)
+                            cur.execute("UPDATE freepacks SET remaining = remaining - 1 WHERE userid = %s AND boostername = %s", [tags['user-id'], args[1]])
+                            self.message(channel, "%s, you open a free %s booster: %s/booster?user=%s" % (tags['display-name'], result[1], config["siteHost"], sender), isWhisper)
```

Comment:
Wording: It may be worth reconsidering the word "free".

On one hand, it makes clear that no money has been spent. On the other hand, it may feel like they might be getting less value, similar to how people seemed to need to be reaffirmed that bet packs contribute towards pity.